config/crd/bases/k8ssandra.io_k8ssandraclusters.yaml

```diff
@@ -1138,270 +1138,2359 @@ spec:
                         type: object
                       dseYaml:
                         properties:
-                          foo:
-                            description: TODO
-                            type: string
-                        type: object
-                      jvmOptions:
-                        properties:
-                          additionalOptions:
-                            description: Additional, arbitrary JVM options (advanced).
-                            items:
-                              type: string
-                            type: array
-                          cassandra_available_processors:
-                            description: 'Available CPU processors. Disabled by default.
-                              Cass Config Builder: supported for Cassandra 3.11 in
-                              jvm.options. Cass Config Builder: supported for Cassandra
-                              4.0 in jvm-server.options. Corresponds to: -Dcassandra.available_processors.'
+                          advanced_replication_options:
```

Comment:
How was this generated? I'm not seeing all dse.yaml options here that are documented, for example `kmip_hosts.key_cache_update_millis` is missing.

Reply:
This was generated manually from the following sources:

1. Documentation, default values and grouping come mostly from the official docs: https://docs.datastax.com/en/dse/6.8/dse-dev/datastax_enterprise/config/configDseYaml.html
2. Typing and structure come from config-builder: https://github.com/datastax/cass-config-definitions/blob/0a4cf53535f56d409e705e5143dba16d5630750d/resources/dse-yaml/dse/dse-yaml-dse-6.8.0.edn
3. Fields that were added in versions > 6.8.0 come from https://github.com/datastax/cass-config-definitions/blob/945606f7dd86b5a6133a1bb02ba0bdd5d61a2876/resources/dse-yaml/dse/dse-yaml-dse-transforms.edn

Also key_cache_update_millis is [here](https://github.com/adutra/k8ssandra-operator/blob/86774b64026d5e7cfd057c4f2c81b7a959529d6d/apis/k8ssandra/v1alpha1/dseconfig_types.go#L759).

Reply:
https://github.com/burmanm/definitions-parser/ would show how to auto-generate from those .edn files, no need to manually do that. 

But I'm not sure if that's all the keys - generating from Java files might make more sense.